packages/text/src/react/heading.tsx

```diff
@@ -0,0 +1,47 @@
+import { compose, css } from 'glamor'
+import React, { HTMLAttributes } from 'react'
+import { useTheme, names } from '@pluralsight/ps-design-system-theme'
+
+import stylesheet from '../css'
+import * as vars from '../vars'
+
+const style = ({ themeName, size }: { themeName?: keyof typeof names; size?: keyof typeof vars.headingSizes }) =>
+  compose(
+    css(stylesheet['.psds-text__heading']),
+    css(stylesheet[`.psds-text__heading.psds-theme--${themeName}`]),
+    css(stylesheet[`.psds-text__heading--size-${size}`]),
+    css(
+      stylesheet[`.psds-text__heading--size-${size}.psds-theme--${themeName}`]
+    )
+  )
+
+interface HeadingStatics {
+  sizes: typeof vars.headingSizes
+}
+
+interface HeadingProps extends HTMLAttributes<HTMLHeadingElement> {
+  size?:  keyof typeof vars.headingSizes
+}
+
+const Heading: React.FC<HeadingProps> & HeadingStatics = ({children, size, ...props}) => {
+  const themeName = useTheme()
+
+  return React.cloneElement(React.Children.only(children as React.ReactElement), {
```

Comment:
```suggestion
  if(!React.isValidElement(children)) return
  
  return React.cloneElement(React.Children.only(children), {
```

i think that should work....